src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -603,8 +603,21 @@ def _report_single_requirement_conflict(
 
         cands = self._finder.find_all_candidates(req.project_name)
         skipped_by_requires_python = self._finder.requires_python_skipped_reasons()
-        versions = [str(v) for v in sorted({c.version for c in cands})]
 
+        is_yanked: Dict[bool, Set[CandidateVersion]] = {True: set(), False: set()}
+        for c in cands:
+            is_yanked[c.link.is_yanked if c.link else False].add(c.version)
+
+        versions = sorted(str(v) for v in is_yanked[False])
+        yanked_versions = sorted(str(v) for v in is_yanked[True])
```

Comment:
We should sort the versions, *then* convert to str for the sorting to be accurate.

Reply:
great catch! could you have a look again?